### DISCUSSION TITLE
Integrate Opentracing traces with opentelemetry traces

### DISCUSSION THREAD

**Discussion by pdkgamage** [open-telemetry/opentelemetry-java#4417] [Q&A] [answered]
Hi All,

There are two micro services.

1. First micro service publishing traces using opentracing
2. Second micro service(implemented using vert.x) publishing traces using opentelemetry.

First micro service sending span-id and trace-id as header parameters. Is it possible to make parent span using those parameters In side the second micro service or any possible solution?

This is how registering opentelemetry in second micro service.

`SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder().build();
OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
.setTracerProvider(sdkTracerProvider)
.setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
.buildAndRegisterGlobal();

    Vertx vertx = Vertx.vertx(new VertxOptions()
            .setTracingOptions(
                    new OpenTelemetryOptions(openTelemetry)
            ));`

Using following opentelemetry dependency.

`<dependency> 
<groupId>io.vertx</groupId> 
<artifactId>vertx-opentelemetry</artifactId> 
<version>4.2.6</version>
</dependency>`

Thanks,
Pradeep

**Comment by trask** [comment, marked as answer]
hi @pdkgamage!

> First micro service sending span-id and trace-id as header parameters

Check out https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#propagator

In particular, this sounds like it might be `ottrace` format?

If it's not a standard trace propagation format already supported by the OpenTelemetry Java project, you may need to write your own [TextMapPropagator](https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/propagation/TextMapPropagator.java) for your use case.

**Reply by pdkgamage** [reply]
Hi Trask,

I used JaegerPropagator and parentSpanContext get set the values as below in the second micro service.
![image](https://user-images.githubusercontent.com/23649771/165497033-6b4a8482-f7f9-40d3-8301-af237a364264.png)

First micro service sending uber-trace-id as bellow.
"uber-trace-id: 5fdbee04a8ff86a7%3Aa9afa6b5a89378a2%3A5fdbee04a8ff86a7%3A1

but micro services traces are not get integrate. am i missing any thing here?

Thanks,
Pradeep

**Reply by trask** [reply]
hi @pdkgamage, I'm not familiar with `uber-trace-id` format, it sounds like you will need to implement a custom trace propagator to handle it

another option is to ask your upstream services to switch to a standard trace propagation format, e.g. W3C, B3 or Jaeger

**Reply by jkwatson** [reply]
that *is* the Jaeger propagation format. :)
